pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -95,7 +95,12 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 							Namespace: svc.Attributes.Namespace,
 							Shortname: shortName,
 						}
-						out.Table[host] = nameInfo
+						if _, f := out.Table[host]; !f || instance.Endpoint.Locality.ClusterID == node.Metadata.ClusterID {
```

Comment:
For non-cluster local endpoints, could we also create names like `<hostname>.<clustername>.<svc>.<ns>.svc.cluster.local` (similar to [MCS](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api#dns) proposal, besides `cluster.local`->`clusterset.local`)?

Some basic support for multi-cluster stateful set would be nice to have if we want to promote MC to stable.

Reply:
I am a bit concerned about adding a new "api" when we already have a replacement planned (MCS). Maybe we should just implement the MCS one?